lhotse/cut.py

```diff
@@ -2135,6 +2208,51 @@ def perturb_volume(self, factor: float, affix_id: bool = True) -> "MixedCut":
             ],
         )
 
+    def reverb_rir(
+        self,
+        rir_cut: "MonoCut",
+        shift_output: bool = True,
+        normalize_output: bool = True,
+        affix_id: bool = True,
+    ) -> "MixedCut":
+        """
+        Return a new ``MixedCut`` that will convolve the audio with the provided impulse response.
+
+        :param rir_cut: The impulse response to use for convolving.
+        :param shift_output: When true, output will be of same length as input.
+        :param normalize_output: When true, output will be normalized to have energy as input.
+        :param affix_id: When true, we will modify the ``MixedCut.id`` field
+            by affixing it with "_rvb".
+        :return: a modified copy of the current ``MixedCut``.
+        """
+        # Pre-conditions
+        assert (
+            self.has_recording
+        ), "Cannot apply reverberation on a MixedCut without Recording."
+        if self.has_features:
+            logging.warning(
+                "Attempting to reverberate a MixedCut that references pre-computed features. "
+                "The feature manifest will be detached, as we do not support feature-domain "
+                "reverberation."
+            )
+            self.features = None
```

Comment:
I don't think this code will work, IIRC `features` is not defined for MixedCut (but `has_features` and `load_features` are). Instead, rely on the implementation inside MonoCut.

Reply:
I see the whole implementation is copied from MonoCut -- this is not correct. MixedCut should just apply the transform to every underlying track and return a new MixedCut.

However, if you're transforming the time information somehow, you might need to adjust the offsets in MixTracks. Probably it's not worth transforming time information to avoid the complexity.

Reply:
I think reverb should be applied on the mixed audio instead of the underlying tracks?

Reply:
Convolutions are linear so it shouldn't matter too much, unless I'm missing something here?

Reply:
It won't matter when you are mixing 2 speech sources, but when you are mixing a speech source and a noise source, you wouldn't want to reverberate the noise source.

Reply:
But in that case you wouldn't call `cut.mix(noise_cut).reverb_rir()` (which is the case being implemented in MixedCut's `reverb_rir`); you'd be calling `cut.reverb_rir().mix(noise_cut)`, and that's the case for MonoCut's `reverb_rir`.

Reply:
Hmm, yeah that's true. I'll make the change accordingly.

Reply:
Also if you want reverb to be applied on the mixed audio, it might be best to run `compute_and_store_recording()` on a `MixedCut` first; or just use `ReverbWithImpulseResponse` transform directly on `mixed_cut.load_audio()`